ygen/ir.go

```diff
@@ -338,23 +358,65 @@ const (
 
 // YANGNodeDetails stores the YANG-specific details of a node
 // within the schema.
+// TODO(wenbli): Split this out so that parts can be re-used by
+// ParsedDirectory.
 type YANGNodeDetails struct {
 	// Name is the name of the node from the YANG schema.
 	Name string
 	// Defaults represents the 'default' value(s) directly
 	// specified in the YANG schema.
 	Defaults []string
-	// Module stores the name of the module that instantiates
-	// the node.
-	Module string
-	// Path specifies the absolute YANG schema node path.
+	// BelongingModule is the module in whose namespace the node belongs.
+	BelongingModule string
+	// RootModule is the module in which the YANG tree the node is attached
+	// to was instantiated (rather than the module that has the same
+	// namespace as the node).
+	//
+	// In this example, leaf has a RootModule of "openconfig-simple", while a
+	// BelongingModule of "openconfig-augment". The DefiningModule is
+	// "openconfig-augment".
+	//
+	//   module openconfig-augment {
+	//     import openconfig-simple { prefix "t"; }
+	//     augment "/t:parent/child/state" {
+	//       leaf zero { type string; }
+	//     }
+	//   }
+	RootModule string
+	// DefiningModule is the module that contains the textual definition of
```

Comment:
Indeed it would return `a` in this case. I'll change to "text" instead of "textual" to be clearer. I've modified an integration test to demonstrate how these three can be different from one another:
https://github.com/openconfig/ygot/blob/927d94dcf31bd2c4126d0f06fb48ab7fe0b2bc7a/testdata/modules/openconfig-simple-augment2.yang#L1-L12
https://github.com/openconfig/ygot/blob/927d94dcf31bd2c4126d0f06fb48ab7fe0b2bc7a/testdata/modules/openconfig-simple-grouping.yang#L1-L9
https://github.com/openconfig/ygot/blob/927d94dcf31bd2c4126d0f06fb48ab7fe0b2bc7a/ygen/genir_test.go#L148-L150

Reply:
Thanks - this is clearer to me. I think `InstantiatingModule` might be clearer for the `BelongingModule` and then `AbsolutePath` per the suggestion in #657 where there is a namespace qualified path (schema) would be the way to get `RootModule`.

I suggest instantiating as this is what rfc7950 calls what a `uses` statement does for a grouping.

Reply:
True, I named it `BelongingModule` because I didn't really think of augmenting as instantiating, but I agree you could look at it that way since you can only augment already-instantiated nodes. Belonging is also more vague than instantiating.

Since I already named "ΛBelongingModule" in the other PR (whose effects have now been reverted) do you mind if I create a separate PR to change both this and that to InstantiatingModule? This would also matches goyang's function of the same name.

Reply:
I'm still not quite sure about this -- if you augment a node, it's still not exactly the same as instantiating -- more like you're adding something to an instantiated node. I'm just not sure if upon first reading `ΛInstantiatingModule()` is really clear about augments either. I can argue that `ΛBelongingModule()` is more apt because the "namespace" is what makes a node belong to some module. What do you think?

Reply:
Ok added PR here: https://github.com/openconfig/ygot/pull/659

Reply:
Sounds great :-)